### ISSUE TITLE
editor.selection not updating after insertText

### ISSUE THREAD

**Issue by codewitch** [ianstormtaylor/slate#4541]
**Description**

Prior to 0.66.0 editor.selection was updating after insertText to account for the next text. However, any text entry after the first does not update the selection and the offset stays at the previous. 
 

**Recording**
I tested on the Link slate example by adding `console.log(editor.selection)` before and after [this line](https://github.com/ianstormtaylor/slate/blob/main/site/examples/links.tsx#L49).
Below is the console logs after inserting 3 `f`s
<img width="1440" alt="Screen Shot 2021-09-22 at 6 26 11 PM" src="https://user-images.githubusercontent.com/5115188/134442411-735c7587-7dea-4835-b0a3-6c4e5134beca.png">

**Sandbox**
I made a simplified sandbox of it showing the console logs of the editor selection
https://codesandbox.io/s/slate-reproductions-forked-39mx5?file=/index.js

**Steps**
To reproduce the behavior:
1. go to above sandbox 
2. insert text in slate editor
3. see the console and see that the editor selection stays the same.
4. change slate versions to 0.65 and repeat
5. see that editor selection updates

**Expectation**
I expected the editor selection to update after `insertText` because that it how it used to work before the update.

**Environment**
- Slate Version: 0.66
- Operating System: macOS
- Browser: Chrome

**Comment by nemanja-tosic** [comment]
Introduced by #3888 - if only insert_text ops are present, editor state lags until `onInput` is called.

**Comment by codewitch** [comment]
Was that and intentional change or a bug that will get fixed?

If it is an intentional change, how do we consistent editor.selection state in `insertText`?

> Introduced by #3888 - if only insert_text ops are present, editor state lags until `onInput` is called.

**Comment by nemanja-tosic** [comment]
I don't think it was intentional, and i see how it can be problematic. I was reporting my findings to start a conversation around it.

**Comment by codewitch** [comment]
Thanks! Was just making sure. What is the process of getting this fixed? Does it need more thumbs ups?

**Comment by dylans** [comment]
@codewitch Thanks. @nemanja-tosic and I have been discussing it and trying to come up with a good solution and we haven't come up with something easy yet or we would have already fixed it. We should also see if @bkrausz has thoughts.

Mostly we just need time to figure out a good solution that hopefully doesn't break something else along the way.

**Comment by NicerDicerPro** [comment]
Hi @dylans,
im currently facing the same issue in version 0.86.0.

Are there any updates on this? 
Should this already be fixed?
